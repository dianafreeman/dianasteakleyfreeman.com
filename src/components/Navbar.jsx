import React from 'react'
import { Link } from 'gatsby'


const Navbar = () => (
  <nav className="navbar is-transparent pin-t">
    <div className="container">
      <div className="navbar-brand">
        <Link to="/" className="navbar-item">
          <figure className="image">{/* <img src={logo} alt="Kaldi" style={{ width: '88px' }} /> */}</figure>
        </Link>
      </div>
      <div className="navbar-start">
        <Link className="navbar-item">What's on this page"</Link>
      </div>
      <div className="navbar-end">
        <a
          className="navbar-item"
          href="https://github.com/GatsbyCentral/gatsby-starter-wordpress"
          target="_blank"
          rel="noopener noreferrer"
        >
          <span className="icon">{/* <img src={github} alt="Github" /> */}</span>
        </a>
      </div>
    </div>
  </nav>
)

export default Navbar
