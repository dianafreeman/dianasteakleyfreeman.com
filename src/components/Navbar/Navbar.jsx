/* eslint-disable react/prop-types */
import React, { useState, useRef } from 'react';
import { FaBars, FaLink } from 'react-icons/fa';
import styled from 'styled-components';
import tw from 'tailwind.macro';
import { useTransition, useSpring, useChain, config, animated } from 'react-spring';
import { Link } from 'gatsby';
import data from '../../config/web/routes';

const MenuContainer = styled.nav`
  ${tw`w-8/9`}
  display: flex;
  flex-wrap: row;
  fontofacolor: black;
  justify-content: center;
  &::before {
    position: absolute;
    width: 100vw;
    height: 3em;
    content: '';
    background-color: white;
  }
`;
const MenuItem = styled(Link)`
  ${tw` mx-6 my-2 py-2 font-bold`}
  position: relative;
  display: block;
`;

const Navbar = () => {
  return (
    <MenuContainer>
      {data.map((item, idx) => (
        <MenuItem key={`nav-${idx}`}>{item.name}</MenuItem>
      ))}
    </MenuContainer>
  );
};
export default Navbar;
