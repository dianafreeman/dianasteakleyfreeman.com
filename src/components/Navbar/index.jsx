import React from "react";

const navigation = [
  { name: "Product", href: "#" },
  { name: "Features", href: "#" },
  { name: "Marketplace", href: "#" },
  { name: "Company", href: "#" },
];

export default function Navbar() {
  return (
    <div>
      <div className="hidden md:block md:ml-10 md:pr-4 md:space-x-8">
        {navigation.map((item) => (
          <a
            key={item.name}
            href={item.href}
            className="font-medium text-gray-500 hover:text-gray-900"
          >
            {item.name}
          </a>
        ))}
        <a
          href="#"
          className="font-medium text-indigo-600 hover:text-indigo-500"
        >
          Log in
        </a>
      </div>
    </div>
  );
}
