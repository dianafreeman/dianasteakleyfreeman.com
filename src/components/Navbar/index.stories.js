import React from "react";
import NavBar from ".";

export const Main = (args) => {
  return <NavBar {...args} />;
};

export default {
  title: "Components/NavBar",
  component: NavBar,
  args: {},
};
