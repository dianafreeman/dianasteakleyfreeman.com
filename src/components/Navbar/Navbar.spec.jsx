import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Navbar from '.';

describe('Navbar', () => {
  it('should render without exploding', () => {
    expect(render(<Navbar />)).not.to.throw;
  });

  it('should match snapshot', () => {
    const tree = renderer.create(<Navbar />).toJSON();
    jestExpect(tree).toMatchSnapshot();
  });
});
