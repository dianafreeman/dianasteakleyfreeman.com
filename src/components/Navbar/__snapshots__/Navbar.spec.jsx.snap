// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: row;
  -ms-flex-wrap: row;
  flex-wrap: row;
  fontofacolor: black;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0::before {
  position: absolute;
  width: 100vw;
  height: 3em;
  content: '';
  background-color: white;
}

.c1 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  font-weight: 700;
  position: relative;
  display: block;
}

<nav
  className="c0"
>
  <a
    className="c1"
  >
    Projects
  </a>
  <a
    className="c1"
  >
    Bio
  </a>
  <a
    className="c1"
  >
    Blog
  </a>
  <a
    className="c1"
  >
    Connect
  </a>
</nav>
`;
